testutils/src/main/java/org/zaproxy/zap/testutils/AlertReferenceError.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2023 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.testutils;
+
+import java.util.function.Function;
+
+class AlertReferenceError {
+
+    enum Cause {
+        INVALID_URI(e -> "Invalid URI: '" + e.reference + "'. Reason: " + e.detail),
+        NOT_HTTPS(e -> "Not HTTPS: " + e.reference),
```

Comment:
Should we enforce HTTPS?

Reply:
Validation in place as it's going to be enabled gradually, as the scan rules are cleaned up.